motr/m0crate/btree_without_tx_seg.c

```diff
@@ -247,8 +270,7 @@ static bool btree_backlink_invariant(const struct m0_be_btree_backlink *h,
 		_0C(m0_cookie_dereference(&h->bli_tree, &cookie) == 0) &&
 		_0C(m0_be_seg_contains(seg, cookie)) &&
 		_0C(parent == cookie) &&
-		_0C(M0_BBT_INVALID < h->bli_type && h->bli_type < M0_BBT_NR) &&
-		_0C(h->bli_gen == seg->bs_gen);
+		_0C(M0_BBT_INVALID < h->bli_type && h->bli_type < M0_BBT_NR);
```

Comment:
Would it be difficult to create a dummy m0_be_seg structure, and point every btree node to it? I'd like to keep code as much in sync with the original, as possible.

Reply:
Re added code for seg. It will be responsibility of btree interface user to pass a dummy m0_be_seg while creating tree. 
For eg. m0crate will add dummy seg to btree while tree creation and same dummy seg will be used for further node creation.